### ISSUE TITLE
False positive errors for functions guards.

### ISSUE THREAD

**Issue by skalidindi** [palantir/tslint#781]
Example: 

```ts
    function isCat(a: any): a is Cat {
         return a.name === 'kitty';
    }
```

This function gives me errors such as block empty, missing white space and missing semi colon.

**Comment by adidahiya** [comment]
which version of TSLint are you using?

**Comment by skalidindi** [comment]
My mistake, it works using the @next branch.
